File: LLM-Translations/GPT-4/codenet/Python/Go/s315467749.go
package main

import (
	"fmt"
)

func main() {
	var n, q, ans int
	fmt.Scan(&n)
	S := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&S[i])
	}
	fmt.Scan(&q)
	T := make([]int, q)
	for i := 0; i < q; i++ {
		fmt.Scan(&T[i])
	}

	ans = 0
	for _, i := range T {
		if i == S[n-1] || i == S[0] {
			ans++
		} else {
			left := 0
			right := n - 1
			for right-left > 1 {
				mid := (left + right) / 2
				if S[mid] == i {
					ans++
					break
				} else if S[mid] < i {
					left = mid
				} else {
					right = mid
				}
			}
		}
	}

	fmt.Println(ans)
}

// 

File: LLM-Translations/GPT-4/codenet/Python/C/s315467749.c
#include <stdio.h>

int main() {
    int n;
    scanf("%d", &n);
    int S[n];
    for (int i = 0; i < n; i++) {
        scanf("%d", &S[i]);
    }

    int q;
    scanf("%d", &q);
    int T[q];
    for (int i = 0; i < q; i++) {
        scanf("%d", &T[i]);
    }

    int ans = 0;
    for (int i = 0; i < q; i++) {
        if (T[i] == S[n - 1] || T[i] == S[0]) {
            ans++;
        } else {
            int left = 0;
            int right = n - 1;
            while (right - left > 1) {
                int mid = (left + right) / 2;
                if (S[mid] == T[i]) {
                    ans++;
                    break;
                } else if (S[mid] < T[i]) {
                    left = mid;
                } else {
                    right = mid;
                }
            }
        }
    }

    printf("%d\n", ans);
    return 0;
}
// 
